File: src/client/java/com/mod/client/compat/ContainerCompat.java
package com.mod.client.compat;

import net.minecraft.client.Minecraft;
import net.minecraft.world.inventory.ContainerInput;

public class ContainerCompat {

    public static void leftClick(int containerId, int slot) {
        Minecraft mc = Minecraft.getInstance();
        if (mc.player == null || mc.gameMode == null) return;
        mc.gameMode.handleContainerInput(
                containerId,
                slot,
                0,
                ContainerInput.PICKUP,
                mc.player
        );
    }

    private ContainerCompat() {}
}

File: src/client/kotlin/com/mod/client/compat/ContainerCompat.kt
package com.mod.client.compat

import net.minecraft.client.Minecraft
import net.minecraft.world.inventory.ContainerInput

object ContainerCompat {
    @JvmStatic
    fun leftClick(containerId: Int, slot: Int) {
        val mc = Minecraft.getInstance()
        val player = mc.player ?: return
        val gameMode = mc.gameMode ?: return
        gameMode.handleContainerInput(
            containerId,
            slot,
            0,
            ContainerInput.PICKUP,
            player
        )
    }
}
